src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -299,6 +300,8 @@ export class SavedObjectsRepository {
       const requiresNamespacesCheck =
         method === 'index' && this._registry.isMultiNamespace(object.type);
 
+      if (object.id == null) object.id = uuid.v1();
```

Comment:
Do we want a `v1` or a `v4` here? The difference is minimal but there are subtle pros and cons for each of them.

EDIT: this was previously done in `generateRawId` which uses a `v1`, so we probably want to keep it that way.